### DISCUSSION TITLE
Using PiecewiseLinear interpolated function in phase-field kernel

### DISCUSSION THREAD

**Discussion by kunokchang** [idaholab/moose#16286] [Q&A Modules: Phase field] [answered]
Dear I would like to call Fermi-Dirac function in phase-field kernel

We have the Fermi-Dirac function in csv file so we built the function as below:

[Functions]
   [./fermi]
      type = PiecewiseLinear
      data_type = fermi.csv
      format = columns
    [../]
[]

but still not clear how to call it in kernel. 

Could you give me any comment on this?

Thank you in advance.

**Comment by laagesen** [comment, marked as answer]
Hi Kunok,

Just to make sure we're on the same page, you're wanting to use the Fermi-Dirac distribution function as shown below, correct?

https://www.doitpoms.ac.uk/tlplib/semiconductors/fermi.php

If this is correct, I would suggest it's easier to code the analytical form of the distribution function into your kernel rather than reading in the function from a csv file, since the analytical form is pretty simple.

**Comment by dschwen** [comment]
Duplicate of #16297
